packages/runtime/container-runtime/src/gc/gcDefinitions.ts

```diff
@@ -232,6 +232,50 @@ export interface IGarbageCollectorCreateParams {
 	readonly getContainerDiagnosticId: () => string;
 }
 
+/**
+ * The configurations for Garbage Collector that determines what runs and how.
+ */
+export interface IGarbageCollectorConfigs {
+	/**
+	 * Tracks if GC is enabled for this document. This is specified during document creation and doesn't change
+	 * throughout its lifetime.
+	 */
+	gcEnabled: boolean;
+	/**
+	 * Tracks if sweep phase is enabled for this document. This is specified during document creation and doesn't change
+	 * throughout its lifetime.
+	 */
+	sweepEnabled: boolean;
+	/**
+	 * Tracks if GC should run or not. Even if GC is enabled for a document (see gcEnabled), it can be explicitly
+	 * disabled via runtime options or feature flags.
+	 */
+	shouldRunGC: boolean;
+	/**
+	 * Tracks if sweep phase should run or not. Even if the sweep phase is enabled for a document (see sweepEnabled), it
+	 * can be explicitly disabled via feature flags. It also won't run if session expiry is not enabled.
+	 */
+	shouldRunSweep: boolean;
+	/** The time in ms to expire a session for a client for gc. */
+	sessionExpiryTimeoutMs: number | undefined;
+	/** The time after which an unreferenced node is ready to be swept. */
+	sweepTimeoutMs: number | undefined;
+	/** The time after which an unreferenced node is inactive. */
+	inactiveTimeoutMs: number;
+	/** Tracks whether GC should run in test mode. In this mode, unreferenced objects are deleted immediately. */
+	testMode: boolean;
+	/**
+	 * Tracks whether GC should run in tombstone mode. In this mode, sweep ready objects are marked as tombstones.
+	 * Tombstone objects behave as if they are deleted but are present for recovery purposes.
```

Comment:
This is an over-simplification.  It's kind of 50-50 IMO.  e.g. Summarizer ignores Tombstone state altogether, and actual behavior for interactive clients is configurable.  Might be good to document this in a markdown file, then we could link to it from here or something.

Reply:
updated comment.